### PR TITLE
Move xds tests to xds_test package and split out NewFakeDiscoveryServer

### PR DESCRIPTION
**Pull request by howardjohn**
This refactors all tests that use NewFakeDiscoveryServer into xds_test instead of xds package. In general, IMO, its best to use the black box test mode in general, to avoid testing low level details and instead target things at a higher level. In particular, when using the fake DS, which is intended for a high level. However, the primary motivation for this is for a future change which moves NewFakeDiscoveryServer out of the production code. This removes quite a few dependencies on our prod binaries and is overall cleaner. This work will be done in a followup, as this PR is fairly large (but should be mostly trivial refactorings).

edit: actually decided it was less noisy to just move NewFakeDiscoveryServer as part of this as well.

### CODE REVIEW

File: pilot/test/xdsfake/fake.go
Comment (hzxuzhonghu) [inline pilot/test/xdsfake/fake.go:1]: nit: xdsfake looks redundant, pilot/test/xdsfake/fake.go =>  pilot/test/xds/fake.go

**Review by hzxuzhonghu — APPROVED** [review]
Good cleanup

**Review by kfaseela — APPROVED** [review]

**Comment by kfaseela** [comment]
when you resolve the conflicts, probably you can take care of Zhonghu's last comment as well

File: pilot/test/xdsfake/fake.go
Comment (howardjohn) [reply pilot/test/xdsfake/fake.go:1]: Some packages will import `pilot/pkg/xds` and `pilot/test/xds` and need to rename the `test/xds` to `xdsfake` anyways, but most don't .I made the change

**Pull request merged**